ExampleMod/Content/NPCs/ExampleBoneMerchant.cs

```diff
@@ -0,0 +1,163 @@
+using ExampleMod.Content.Dusts;
+using ExampleMod.Content.Items;
+using System.Linq;
+using Terraria;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.ModLoader;
+using Terraria.Utilities;
+using Terraria.GameContent.Bestiary;
+
+namespace ExampleMod.Content.NPCs
+{
+	/// <summary>
+	/// The main focus of this NPC is to show how to make something similar to the vanilla bone merchant;
+	/// which means that the NPC will act like any other town NPC but won't have a happiness button, won't appear on the minimap,
+	/// and will spawn like an enemy NPC. If you want a traditional town NPC instead, <see cref="ExamplePerson"/>.
+	/// </summary>
+	public class ExampleBoneMerchant : ModNPC
+	{
+		public override void SetStaticDefaults() {
+			Main.npcFrameCount[Type] = 25; // The amount of frames the NPC has
+
+			NPCID.Sets.ExtraFramesCount[Type] = 9; // Generally for Town NPCs, but this is how the NPC does extra things such as sitting in a chair and talking to other NPCs.
+			NPCID.Sets.AttackFrameCount[Type] = 4;
+			NPCID.Sets.DangerDetectRange[Type] = 700; // The amount of pixels away from the center of the npc that it tries to attack enemies.
+			NPCID.Sets.AttackType[Type] = 0;
+			NPCID.Sets.AttackTime[Type] = 90; // The amount of time it takes for the NPC's attack animation to be over once it starts.
+			NPCID.Sets.AttackAverageChance[Type] = 30;
+			NPCID.Sets.HatOffsetY[Type] = 4; // For when a party is active, the party hat spawns at a Y offset.
+
+			// Influences how the NPC looks in the Bestiary
+			NPCID.Sets.NPCBestiaryDrawModifiers drawModifiers = new NPCID.Sets.NPCBestiaryDrawModifiers(0) {
+				Velocity = 1f, // Draws the NPC in the bestiary as if its walking +1 tiles in the x direction
+				Direction = 1 // -1 is left and 1 is right. NPCs are drawn facing the left by default but ExamplePerson will be drawn facing the right
+			};
+
+			NPCID.Sets.NPCBestiaryDrawOffset.Add(Type, drawModifiers);
+		}
+
+		public override void SetDefaults() {
+			NPC.friendly = true; // NPC Will not attack player
+			NPC.width = 18;
+			NPC.height = 40;
+			NPC.aiStyle = 7;
+			NPC.damage = 10;
+			NPC.defense = 15;
+			NPC.lifeMax = 250;
+			NPC.HitSound = SoundID.NPCHit1;
+			NPC.DeathSound = SoundID.NPCDeath1;
+			NPC.knockBackResist = 0.5f;
+
+			AnimationType = NPCID.Guide;
+		}
+
+		//This hook is the most important part of this NPC. Since it is true, it tells the game that we want this NPC to act like a town NPC without being one.
+		//What that means is: the NPC will have the AI of a town NPC, will attack like a town NPC, and have a shop (or any other additional functionality if you wish) like a town NPC.
+		//However, the NPC will not have their head displayed on the map, will de-spawn when no players are nearby or the world is closed, will spawn like any other NPC, and have no happiness button when chatting.
+		public override bool ActsLikeATownNPC() {
```

Comment:
Please convert both this and SpawnsWithCustomName to be properties instead of methods.
There doesn't look like any need for fully fleshed out virtual methods here.